### PR TITLE
Merge k8s and go updates

### PR DESCRIPTION
**Pull request by dholbach**
This PR subsumes #132 , #202 and #231 to make testing a bit more straight-forward and because dependabot was falling over trying to rebase.

Closes: #132, #202, #231 .

### CODE REVIEW

**Comment by evrardjp** [comment]
Wow this is the big deal! :)

File: README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ server:
 
 | kured  | kubectl | k8s.io/client-go | k8s.io/apimachinery | expected kubernetes compatibility |
 |--------|---------|------------------|---------------------|-----------------------------------|
-| master | 1.18.8  | v0.18.8          | v0.18.8             | 1.17.x, 1.18.x, 1.19.x            |
+| master | 1.19.4  | v0.19.4          | v0.19.4             | 1.17.x, 1.18.x, 1.19.x            |
```
Comment (dholbach) [inline README.md:48]: There is no `kindest/node` image for 1.20 yet - maybe unofficially somewhere? But we can start testing locally I guess.

Comment (dholbach) [reply README.md:48]: Just pointing this out here as to why I didn't update the "expected k8s compatibility" section. We can do that in the next PR.

Comment (evrardjp) [reply README.md:48]: Yeah we can probably build one. Or maybe we should wait for the official release before working on 1.20? We are so much ahead, it's impressive :)

Comment (evrardjp) [reply README.md:48]: Shouldn't we edit this to 1.17.x,1.18.x,1.19.x?

**Review by evrardjp — CHANGES_REQUESTED** [review]
It's good for me, I would just change that line in the README.

**Pull request merged**